pkg/config/piped.go

```diff
@@ -256,6 +261,59 @@ type HelmChartRepository struct {
 	Password string `json:"password"`
 	// Whether to skip TLS certificate checks for the repository or not.
 	Insecure bool `json:"insecure"`
+
+	// Remote address of the Git repository used to clone Helm charts.
+	// e.g. git@github.com:org/repo.git
+	GitRemote string `json:"gitRemote"`
+	// The path to the private ssh key file used while cloning Helm charts from above Git repository.
+	SSHKeyFile string `json:"sshKeyFile"`
+}
+
+func (r *HelmChartRepository) IsHTTPRepository() bool {
+	return r.Name != "" && r.Address != ""
+}
+
+func (r *HelmChartRepository) IsGitRepository() bool {
+	return r.GitRemote != ""
+}
+
+func (r *HelmChartRepository) Validate() error {
+	if r.IsHTTPRepository() {
+		if r.Name == "" {
+			return errors.New("name must be set")
+		}
+		if r.Address == "" {
+			return errors.New("address must be set")
+		}
+	}
+
+	if r.IsGitRepository() {
+		if r.GitRemote == "" {
+			return errors.New("gitRemote must be set")
+		}
+	}
+
+	return errors.New("either HTTP repository or Git repository must be configured")
```

Comment:
I think this is not right, since in case both or one of HTTP or GitRemote pattern are provided, the error will be returned too. We may need to add a return after L293 or change the implementation.

Reply:
Get your point. Fixed it!

Reply:
Or should we add a `type` field, which should be either `HTTP` or `GIT`?

Reply:
That's nice too 👍 
Just want to confirm one point: do we only allow one of those types to be configured, not both at the same configure, right?

Reply:
In such case, the current implementation is LGTM, may be we don't need to add type to keep the configuration simple 😄

Reply:
> do we only allow one of those types to be configured, not both at the same time for one chartRepo, right?

Yes. that is right.

Reply:
>we only allow one of those types to be configured

I'd say we should take this one, so I feel like just adding this at first line within this function is enough
```go
if r.IsHTTPRepository() && r.IsGitRepository() {
    return errors.New("Only one of HTTP or GIt can be set")
}

Reply:
Ah sure, nice catch 👍 Don't think this thing is that complicated at first look.

Reply:
Well, this validation is a little bit odd to me 🤔 , since we currently determine whether it's Git or HTTP based on whether each field (Name, Address, GitRemote) is set, rather than by `type` field.
I mean absolutely nothing can fail this validation.

Reply:
If you expect the logic inside `IsHTTPRepository` and `IsGitRepository` to change in the future, that woks fine 👍

Reply:
> I mean absolutely nothing can fail this validation.

I feel you 😂 Maybe we only need
```golang
if r.IsHTTPRepository() && r.IsGitRepository() {
  return errors.New("Only one of HTTP or GIt can be set")
}

if r.IsHTTPRepository() {
  return nil
}

if r.IsGitRepository() {
  return nil
}

return error.New("One of required")
```

Reply:
That's exactly what we need, at least now 👍

Reply:
I see. Just feel weird when seeing this check. 

``` golang
if r.IsHTTPRepository() {
  return nil
}

if r.IsGitRepository() {
  return nil
}
```

Reply:
If the logic under `IsHTTPRepository` only uses, for example, `Address` to determine whether this repo configuration is HTTP or not, we may place the name check under the `if r.IsHTTPRepository()` condition, how do you guys think about that?

Reply:
I think we should add `type` field to make think to be simple. 😄

Reply:
In that case, the validation logic is easier for us but looks like it's not simple for our users. Since to keep support for running applications, we need to make the `type` field default to `HTTP`, and if users want to use type `Git` they have to explicitly specify that value in the configuration. But I don't have a strong opinion on this, just feel like it will be easier for me if I'm going to configure this without the `type` field.

Reply:
Yeah. Exactly as what you said. But the `GIT` type is only needed when user wants to specify a different SSH key for that repository, so I guess that won't cause much trouble for the user.
Btw, I added the `type` field. PTAL.

Reply:
Okay, we may add more types value to that field. In that case, adding type field and making type default to `HTTP` is good for future addition 👍

Reply:
Alright.